tests/test_projection_general.py

```diff
@@ -0,0 +1,42 @@
+from fenics import *
+import numpy as np
+import pytest
+
+import cashocs
+from cashocs._optimization.topology_optimization import bisection
+
+rng = np.random.RandomState(300696)
+
+
+@pytest.fixture
+def levelset_function():
+    mesh, subdomains, boundaries, dx, ds, dS = cashocs.regular_mesh(16)
+
+    CG1 = FunctionSpace(mesh, "CG", 1)
+    psi_exp = Expression("x[0]-0.5", degree=1)
+    psi = Function(CG1)
+    psi.vector()[:] = project(psi_exp, CG1).vector()[:]
+
+    return psi
+
+
+@pytest.mark.parametrize(
+    "volume",
+    [
+        ([rng.rand() / 4.0 + 0.5, rng.rand() / 4.0 + 0.75]),
+        ([rng.rand() / 4.0, rng.rand() / 4.0 + 0.25]),
+        ([rng.rand() / 2.0, rng.rand() / 2.0 + 0.5]),
+        ([rng.rand()]),
+    ],
+)
+def test_projection_method_for_topology_optimization(levelset_function, volume):
+    if len(volume) == 1:
+        target = volume[0]
+    else:
+        target = min(max(volume[0], 0.5), volume[1])
+    projection = bisection.projection_levelset(
+        levelset_function, volume_restriction=volume
+    )
+    projection.project()
+    projection_volume = projection.evaluate(0.0, 0.0)
```

Comment:
Here, I would say it is more sensible to assemble the volume of the levelset function directly - you haven't really checked that projection.evaluate(0,0) actually does return the volume (but this could be another test)

Reply:
I added an additional test which checks that projection.evalutate(0, 0) works properly.